packages/build-tools/src/utils/project.ts

```diff
@@ -119,3 +102,32 @@ async function installDependencies<TJob extends Job>(ctx: BuildContext<TJob>): P
     env: ctx.env,
   });
 }
+
+async function runExpoDoctor<TJob extends Job>(ctx: BuildContext<TJob>): Promise<boolean> {
```

Comment:
a bit of a nitpick, but I don't like when a function that is doing sth also returns information. Based on the name I would not expect runExpoDoctor to return a boolean. If you want to extract part of the code, you could:
- make runExpoDoctor returns SpawnResults
- `catch` and match logic are in code that calls  runExpoDoctor